src/graphql/execution/execute.py

```diff
@@ -459,12 +468,16 @@ async def get_results() -> dict[str, Any]:
                 field = awaitable_fields[0]
                 results[field] = await results[field]
             else:
-                results.update(
-                    zip(
-                        awaitable_fields,
-                        await gather(*(results[field] for field in awaitable_fields)),
-                    )
-                )
+                tasks = {}
+                for field in awaitable_fields:
+                    tasks[create_task(results[field])] = field  # type: ignore[arg-type]
+
+                done, pending = await wait(tasks, return_when=FIRST_EXCEPTION)
```

Comment:
The other approach is to allow all fields to finish executing and then raise the exception.

Reply:
You could do it using gather() like this:

```python
                tasks = [create_task(results[field]) for field in awaitable_fields]
                try:
                    awaited_results = await gather(*tasks)
                except Exception:
                    # cancel all pending tasks before bubbling up the error
                    for task in tasks:
                        if not task.done():
                            task.cancel()
                    await gather(*tasks, return_exceptions=True)
                    raise
                results.update(
                    zip(
                        awaitable_fields,
                        awaited_results,
                    )
                )
```

Reply:
This shows more explicitly what happens in the error case.

But your approach using wait() is also fine. Maybe add a comment before the results.update() mentioning that potential exceptions will be raised at this point (when task.result() is called).